src/components/FeatureViews/AdminPortal/ActionLog.js

```diff
@@ -0,0 +1,482 @@
+                                <Grid>
+                                    Object
+                                </Grid>
+                                <Grid>
+                                    <SwapVertIcon onClick={handleObjectSort} style={{ color: "grey" }} />
+                                </Grid>
+                            </Grid>
+                            <Grid>
+                                <FormControl
+                                    variant="outlined"
+                                >
+                                    <InputLabel shrink={false} style={{ lineHeight: "5px" }} margin="dense" >{objectType ? "" : "Object Type"}</InputLabel>
+                                    <Select
+                                        id="select"
+                                        labelId="label"
+                                        value={objectType}
+                                        onChange={handleObjectSelection}
+                                        style={{ width: 130 }}
+                                        classes={{
+                                            outlined: classes.outlined,
+                                            root: classes.root,
+                                            "MuiInputBase-input": classes.MuiInputBase
+                                        }}
+                                    >
+                                        {objectOptions.map((objectType) =>
+                                            (<MenuItem key={objectType} value={objectType}>
+                                                {objectType}
+                                            </MenuItem>)
+                                        )}
+                                    </Select>
+                                </FormControl>
+                            </Grid>
+                        </TableCell>
+                        <TableCell>
+                            <Grid container>
+                                <Grid>
+                                    Details
+                                </Grid>
+                            </Grid>
+                            <Grid>
+                                <br />
+                            </Grid>
+                        </TableCell>
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {logData && logData.logs.results.map((actionItem) => {
+                        return (
+                            <TableRow>
+                                <TableCell>
+                                    <Moment
+                                        date={actionItem.date}
+                                        format="dddd, M/D/YYYY h:mm a"
+                                    />
+                                </TableCell>
+                                <TableCell>
+                                    {userOptions && userOptions.map((userType) => {
```

Comment:
same thing here - use the `loading` variable to validate that `userOptions` is valid

Reply:
it also looks more like you're using this `.map()` function as a filter function (`if(userTypes.user.id == actionItem.userId)` so do this prior to the return statement instead
```
const selectedUser = userOptions.filter( userType => (userType.user.id == actionItem.userId);
const selectedUserDisplayText = `${fullName(userType.user)} #${actionItem.userId}`;
return ( ...<div>{selectedUserDisplayText}</div>);
```